test/e2e/framework/pod/resource.go

```diff
@@ -445,6 +445,24 @@ func NewAgnhostPod(ns, podName string, volumes []v1.Volume, mounts []v1.VolumeMo
 	return pod
 }
 
+// NewAgnhostPodWithContainer returns a pod with prepare's containers.
+func NewAgnhostPodWithContainer(ns, podName string, volumes []v1.Volume, cs []v1.Container) *v1.Pod {
```

Comment:
this really doesn't belong here ... also, the name is not really `NewAgnhostPodWithContainer` there is no guarantee it uses agnhost , but that is an unrelated comment to the PR.
Move this directly inside `createPodWithContainer`